lib/Default/Plotter.class.inc

```diff
@@ -293,12 +293,13 @@ class Distance {
 		return $this->path[0];
 	}
 	
-	public function followPath($isWarp=false) {
-		if($isWarp===false)
-			$this->distance--;
-		else
+	public function followPath() {
+		$nextSectorID = array_shift($this->path);
+		if (in_array($nextSectorID, array_values($this->warpMap))) {
```

Comment:
I think this can be simplified and made more performant with
```php
isset($this->warpMap[$nextSectorID])

Reply:
Unfortunately, `$nextSectorID` is an array value, not an array key, so this won't work. However, I think we could just flip the way the `warpMap` is constructed (and swap any references to keys <-> values), and then your suggestion would work.

Reply:
Ah, it'd still make sense to use `in_array($nextSectorID, $this->warpMap)` in preference then

Reply:
Good point, I'll make this change now. Thanks!

I see you added the strict comparison since your original comment. Was there a specific reason why you felt this was needed?

Reply:
I think in this case it would probably work without the strict check, but the strict check helps avoid a whole lot of potential weirdness in general so I think it'd just be best practice to use it - rather than potentially having to go through a load of debugging for a really weird error in the future